lib/Frontend/CompilerInvocation.cpp

```diff
@@ -44,6 +44,12 @@ void CompilerInvocation::setMainExecutablePath(StringRef Path) {
   llvm::sys::path::remove_filename(LibPath); // Remove /bin
   llvm::sys::path::append(LibPath, "lib", "swift");
   setRuntimeResourcePath(LibPath.str());
+
+  llvm::SmallString<128> DiagnosticDocsPath(Path);
+  llvm::sys::path::remove_filename(DiagnosticDocsPath); // Remove /swift
+  llvm::sys::path::remove_filename(DiagnosticDocsPath); // Remove /bin
+  llvm::sys::path::append(DiagnosticDocsPath, "share", "doc", "diagnostics");
```

Comment:
We don't do it on Apple OSes, but it *is* possible to install Swift at the root of the filesystem, meaning this directory would literally be `/usr/share/doc/diagnostics`. I think it would be a good idea to put it at a path like `/usr/share/doc/swift/diagnostics` instead.

Reply:
(We should also consider placing these files in the compiler resource directory. That directory is by default computed relative to the executable already, but we have standard mechanisms to override it; I'm not sure the educational notes need a separate location.)

Reply:
I didn't know we supported installing Swift at the root, good to know. By compiler resource directory, do you mean `lib/swift` / `lib/swift_static`? I wasn't sure how to copy the docs to the right place for each configuration and platform.

Reply:
I changed this to copy from `userdocs/diagnostics` in the swift repo to `/usr/share/doc/swift/diagnostics` in the toolchain. I'm still open to putting these in the resource directory if that makes the most sense though.